### PR TITLE
Declared support for Dart 2 and updated to latest standards

### PR DESCRIPTION
**Pull request by pulyaevskiy**

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @yjbanov @cbracken

File: pubspec.yaml [hunk 1]
```diff
@@ -13,7 +13,9 @@ authors:
 - Victor Berchet <victor@suumit.com>
 description: Compute hashCode correctly for your objects
 homepage: https://github.com/QuiverDart/quiver_hashcode
+
 environment:
-  sdk: '>=1.8.0 <2.0.0'
+  sdk: '>=1.8.0 <3.0.0'
```
Comment (yjbanov) [inline pubspec.yaml]: Let's match what `package:quiver` does: https://github.com/google/quiver-dart/blob/master/pubspec.yaml

Specifically, Dart 2 support comes in version 2.0.0 and it also simultaneously drop Dart 1.0.0 support.

Comment (pulyaevskiy) [reply pubspec.yaml]: OK, should I also bump the package version to `2.0.0` than?

Comment (pulyaevskiy) [reply pubspec.yaml]: Sorry for being slow here (traveling). Just pushed an update. PTAL.

**Comment by yjbanov** [comment]
2.0.0 is published to pub. Thanks again!

**Pull request merged**